### PR TITLE
get all content(subhubs, quiz, flashcards) from db to use as data for the type ahead feature

### PR DESCRIPTION
**Pull request by Fnuworsu**
# Description  
- What does this PR do?  
  `Implements a comprehensive search API endpoint that allows users to search across multiple content types (flashcards, quizzes, subhubs, chapters, and posts) with a single query.`  
---

# Milestones / Related Work  
- Feature or user story this contributes to: `Global Search Functionality needed for technical challenge #2`  
---

# Resources and References  
- Tutorials, documentation, or code examples used:  
  `Prisma documentation for search queries`  
---

# Test Plan  
- How was this tested?  
  `Tested using Insomnia with various search queries against development database. Verified results across all content types.`  

## Edge Cases Tested  
*   `Empty search query returns empty result sets`
*   `Case insensitive matching works correctly`
*   `Search across related entities (e.g., finding flashcards by chapter title)`
*   `Proper error handling when database connection fails`

### CODE REVIEW

File: backend/routes/searchRoutes/searchRoutes.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
+const express = require('express')
+const { PrismaClient } = require('@prisma/client')
+const { isAuthenticated } = require('../../middleware/middleware')
+
+const router = express.Router()
+const prisma = new PrismaClient()
+
+// Search across all content types
+router.get('/api/search', isAuthenticated, async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Fnuworsu) [reply backend/routes/searchRoutes/searchRoutes.js:9]: @sinap-fb yes. `isAuthenticated` middleware ensures that
